.github/workflows/main.yml

```
@@ -1,13 +1,16 @@
 name: Publish
 
 on:
-  push:
-    branches: ["main"]
+  workflow_run:
+    workflows: ["Test Commands"]
+    types:
+      - completed
 
 jobs:
   publish:
     name: Publish
     runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'success'  && github.event.workflow_run.head_branch == 'main' }}
```

The `workflow_run` event can filter by branch directly, see [`on.workflow_run.<branches|branches-ignore>`](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#onworkflow_runbranchesbranches-ignore)
```suggestion
  workflow_run:
    workflows: ['Test Commands']
    types: ['completed']
    branches: ['main']

jobs:
  publish:
    name: Publish
    runs-on: ubuntu-latest
    if: ${{ github.event.workflow_run.conclusion == 'success' }}
```